pkg/plugin/v2/scaffolds/internal/templates/boilerplate.go

```diff
@@ -34,19 +34,48 @@ type Boilerplate struct {
 	// License is the License type to write
 	License string
 
+	// Licenses maps License types to their actual string
+	Licenses map[string]string
+
 	// Owner is the copyright owner - e.g. "The Kubernetes Authors"
 	Owner string
 
 	// Year is the copyright year
 	Year string
 }
 
+// Validate implements file.RequiresValidation
+func (f Boilerplate) Validate() error {
+	if f.License == "" {
+		// A default license will be set later
+	} else if _, found := knownLicenses[f.License]; found {
+		// One of the know licenses
+	} else if _, found := f.Licenses[f.License]; found {
+		// A map containing the requested license was also provided
+	} else {
+		return fmt.Errorf("unknown specified license %s", f.License)
+	}
+
+	return nil
+}
+
 // SetTemplateDefaults implements input.Template
 func (f *Boilerplate) SetTemplateDefaults() error {
 	if f.Path == "" {
 		f.Path = filepath.Join("hack", "boilerplate.go.txt")
 	}
 
+	if f.License == "" {
+		f.License = "apache2"
+	}
+
+	if f.Licenses == nil || len(f.Licenses) == 0 {
+		f.Licenses = make(map[string]string, len(knownLicenses))
+		for key, value := range knownLicenses {
+			f.Licenses[key] = value
+		}
+	}
```

Comment:
Probably not necessary, but the following changes makes it so the caller can set custom licenses but still gets other defaults:
```suggestion
	if f.Licenses == nil {
		f.Licenses = make(map[string]string, len(knownLicenses))
	}
	for key, value := range knownLicenses {
		if _, hasLicense := f.Licenses[key]; !hasLicense {
			f.Licenses[key] = value
		}
	}
```